In [6]:
words = open("words.txt").read().splitlines()

First we read the file in a list.
Each element in the list will be a line from the file.
So each element will be a single word in this case.

In [7]:
letters = {}
for word in words:
    key = word[0]
    letters[key] = 1 if key not in letters else letters[key] + 1

We first create a dictionary which will have a letter as a key and the value will be
the amount of times it has been the first letter of a word.

For each word we take the first letter by doing word[0] then we check if the dictionary
contains it, if so we add 1 to the value otherwise we set the value to 1.

In [8]:
letters["a"]

24370

As you can see words.txt contains 24370 words that start with the letter a.